modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -1381,7 +1486,7 @@ def _find_index_or_col(self, col):
 
         if self._index_cols is not None:
             for idx_col in self._index_cols:
-                if re.match(f"__index__\\d+_{col}", idx_col):
+                if col == idx_col or re.match(f"__index__\\d+_{col}", idx_col):
```

Comment:
Isn't we're naming all the index columns like `"__index__{name}"`. In which cases this exception is made?

Reply:
This to allow direct usage of mangled index column names in `sort_rows`. Otherwise we cannot sort by multiple index columns having the same name.

Reply:
got it